### ISSUE TITLE
[Question] Go modules way of handling the Generated files

### ISSUE THREAD

**Issue by shettyh** [golang/go#29294]
### What version of Go are you using (`go version`)?

<pre>
$ go version
go version go1.11.2 linux/amd64
</pre>

### Does this issue reproduce with the latest release?
Yes

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
$ go env

GOARCH="amd64"
GOBIN="/root/shetty/go/bin"
GOCACHE="/root/.cache/go-build"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/root/shetty/go"
GOPROXY=""
GORACE=""
GOROOT="/usr/local/go"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/linux_amd64"
GCCGO="gccgo"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/root/shetty/base_services/telemetry/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build371566123=/tmp/go-build -gno-record-gcc-switches"
</pre></details>

### What did you do?
Not able to find any relevant documents.

Basically i am generating the Protobuf files to `GOPATH/src/github.com/xxx/....` . But i am not able to add this as import in my Go Modules project

Is there any known way to handle this situation ?

**Comment by bcmills** [comment]
If you're using the `go` command, you need to check in all of the relevant generated sources.

**Comment by bcmills** [comment]
See https://blog.golang.org/generate:
> if the containing package is intended for import by `go get`, once the file is generated (and tested!) it must be checked into the source code repository to be available to clients.

**Comment by shettyh** [comment]
@bcmills Thanks for the response

These files are generated at the build time and not committed to VCS.
Is there any way to exclude this in `go.mod` and directly use from $GOPATH ?

If i try to use the `replace` directive in mod file it is checking for `go.mod` file inside generated files directory

**Comment by bcmills** [comment]
> These files are generated at the build time and not committed to VCS.

Is there a fundamental reason why they *cannot* be committed to VCS?

> Is there any way to exclude this in `go.mod` and directly use from $GOPATH ?

No: module mode intentionally avoids the use of modified sources for Go packages, and regenerating a source file is definitely a modification.

> If i try to use the `replace` directive in mod file it is checking for `go.mod` file inside generated files directory

Yes, you may need to run `go mod init` in that directory with an appropriate module path.
